File: 3]. Competitive Programming/LeetCode/Problems/Python/0001)_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}
        
        for i, num in enumerate(nums):
            if target-num in num_to_index:
                return [num_to_index[target-num], i]
            num_to_index[num] = i
        return []

File: 3]. Competitive Programming/08]. LeetCode/1]. Problems/C++/0001)_Two_Sum.cpp
class Solution
{
  public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        unordered_map<int, int> hash;
        for(int i=0; i<nums.size(); i++)
        {
            if(hash.find(target-nums[i]) != hash.end())
                return {hash[target-nums[i]], i};
            hash[nums[i]] = i;
        }
        return {};
    }
};
